newrelic/resource_newrelic_workload.go

```diff
@@ -0,0 +1,327 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/errors"
+	"github.com/newrelic/newrelic-client-go/pkg/workloads"
+)
+
+func resourceNewRelicWorkload() *schema.Resource {
+	userReference := &schema.Schema{
+		Type:     schema.TypeList,
+		MaxItems: 1,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"email": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"gravatar": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"id": {
+					Type:     schema.TypeInt,
+					Computed: true,
+				},
+				"name": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceNewRelicWorkloadCreate,
+		Read:   resourceNewRelicWorkloadRead,
+		Update: resourceNewRelicWorkloadUpdate,
+		Delete: resourceNewRelicWorkloadDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"account": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Computed:    true,
+				Description: "The account the workload belongs to.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"created_at": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The moment when the object was created, represented in milliseconds since the Unix epoch.",
+			},
+			"created_by": userReference,
+			"entity": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
+				Description: "A list of entities manually assigned to this workload.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"guid": {
```

Comment:
Are there any other attributes that will be added under this Set? If not we can simply this whole block to look something like 

```
"entity_guids": {
  Type:     schema.TypeSet,
  Optional: true,
  Elem: &schema.Schema{
  Type: schema.TypeString,
},
```

Reply:
👍